corehq/util/cache_utils.py

```diff
@@ -41,3 +51,24 @@ def should_backoff(cls, key):
 def is_rate_limited(rate_limit_key):
     ExponentialBackoff.increment(rate_limit_key)
     return ExponentialBackoff.should_backoff(rate_limit_key)
+
+
+class ExponentialGrowth(CacheExponential):
+    klass_key = "django-exp-growth"
+
+    EXPONENTIAL_RATE = 2
+    BASE_TIME = 5
+    MAX_TIME = 60
+
+    @classmethod
+    def get_next_time(cls, event_key, base_time=BASE_TIME, max_time=MAX_TIME):
+        if not event_key:
+            return base_time
+        repeat_number = cls.increment(event_key) - 1
+        exponential = cls._exponential(
+            base_time, cls.EXPONENTIAL_RATE, repeat_number)
+        return exponential if exponential < max_time else max_time
```

Comment:
Not sure how frequent this would be but we can avoid calling `_exponential` if the current retry time is already above max time instead of calculating the time and then checking.